napari/layers/intensity_mixin.py

```diff
@@ -43,6 +53,34 @@ def reset_contrast_limits_range(self):
             info = np.iinfo(self.dtype)
             self.contrast_limits_range = (info.min, info.max)
 
+    @property
+    def autoscale_source(self):
+        """Data source for autoscaling, either 'data' (full) or 'slice'."""
+        return self._autoscale_source
```

Comment:
Do we really need both `full` and `slice`. In `full` mode the effect is just to have the clims reset to the same value whenever the slice is changed.

Reply:
> In full mode the effect is just to have the clims reset to the same value whenever the slice is changed.

... assuming layer.data itself isn't changing, yes.

you'd need full mode if `layer.data` was changing and that was important to you.  `slice` mode says you only care about the currently display slice, regardless of the rest of the volume.  I agree slice is more tightly connected to the viewer (and more useful), but if display of a given 2D plane in the context of the full volume was important, then `data` mode is more useful.

happy to get rid of "volume" mode, but curious what @jni wants here

Reply:
> ... assuming layer.data itself isn't changing, yes.

Ah i guess you're thinking about an image acquisition context - ok I missed that, I see how that could be more useful ...

I might rename them `contrast_autoscale_source` and `contrast_autoscale` then

Maybe we still start with only exposing `contrast_autoscale` bool in the GUI though